### ISSUE TITLE
Should Python.Runtime.Py.With use Runtime.None instead of new PyObject(Runtime.PyNone)?

### ISSUE THREAD

**Issue by IvanAntipov** [pythonnet/pythonnet#1192]
> I'm not sure if I'm right, but I apologize in advance if I'm misleading the community

After long exectution (I train large ML model), I fall into ``Fatal Python error: deallocating None``

```
Fatal Python error: deallocating None

Current thread 0x00007fc518efd700 (most recent call first):

Thread 0x00007fc4997fe700 (most recent call first):
  File "/usr/local/lib/python3.7/threading.py", line 300 in wait
  File "/usr/local/lib/python3.7/queue.py", line 179 in get
  File "/usr/local/lib/python3.7/site-packages/tensorboard/summary/writer/event_file_writer.py", line 232 in run
  File "/usr/local/lib/python3.7/threading.py", line 926 in _bootstrap_inner
  File "/usr/local/lib/python3.7/threading.py", line 890 in _bootstrap
```

> Sadly, I can't provide full repro, because problem does not occur in small data set, and I work on private project.

In my code I use [Should Python.Runtime.Py.With](https://github.com/pythonnet/pythonnet/blob/ec424bb0a8f0217c496898395880cf9b99d073e6/src/runtime/pythonengine.cs#L768) method.

In this method I see PyNone to be used without increment of references count

```
IntPtr type = Runtime.PyNone;
...
new PyObject(type)
```

As I undestand from [stackoverflow](https://stackoverflow.com/questions/26742837/creating-a-c-sharp-nullable-int32-within-python-using-python-net-to-call-a-c-s) each usage of None should increment Ref counter.

There is [Python.Runtime.Runtime.None](https://github.com/pythonnet/pythonnet/blob/7a9dcfa162f038fc599988ca59eb49a84be0affc/src/runtime/runtime.cs#L458) property

Should Runtime.None be used instead of Runtime.PyNone

### Environment

-   Pythonnet version: pythonnet_netstandard (I know, it is a fork of this project, but the code under dicussion is under this project) "Python.Runtime.Mono" Version="3.7.1"
-   Python version: 3.7.7
-   Operating System: Ubuntu 18.04.3 (in docker)

**Comment by IvanAntipov** [comment]
The proble with "delallocating None" has gone, after  I have switched to my own implmetation of "Py.With"


F# code:
```fsharp
    let inline pyWith(pyObj: PyObject, func: PyObject -> 'T) =
        let mutable ptr: PyObject option = None 
        let mutable ptr2: PyObject option = None
        let mutable ptr3: PyObject option = None
        let mutable ex : PythonException = null;
        let mutable res : 'T option = None

        try 
            let obj2: PyObject = pyObj.InvokeMethod("__enter__");
            
            res <- Some <| func(obj2)
        with
        | :? PythonException as ex2 ->            
            ex <- ex2;
            ptr <- Some <| PyObject ex.PyType
            ptr2 <- Some <| PyObject ex.PyValue;
            ptr3 <- Some <| PyObject ex.PyTB;

        let toPyNoneIfEmpty(o: PyObject option) = o |> Option.defaultWith(fun () -> Runtime.GetPyNone())

        let exitObject : PyObject = pyObj.InvokeMethod("__exit__", ptr |> toPyNoneIfEmpty, ptr |> toPyNoneIfEmpty, ptr2 |> toPyNoneIfEmpty, ptr3 |> toPyNoneIfEmpty);
        
        if (ex |> isNull |> not && not <| exitObject.IsTrue()) then
            raise ex

        res |> Option.get
```

**Comment by lostmsu** [comment]
@IvanAntipov this has already been fixed before 2.5.0 was released in #1062. The very link you provided to this repo has `XIncref(type)` just a few lines above.
